nexus/db-model/src/region.rs

```diff
@@ -112,4 +121,18 @@ impl Region {
     pub fn deleting(&self) -> bool {
         self.deleting
     }
+
+    /// The size of the Region without accounting for any overhead
+    pub fn requested_size(&self) -> u64 {
+        self.block_size().to_bytes()
+            * self.blocks_per_extent()
+            * self.extent_count()
+    }
+
+    /// The size the Crucible agent would have reserved during ZFS creation,
+    /// which is some factor higher than the requested region size to account
+    /// for on-disk overhead.
+    pub fn reserved_size(&self) -> u64 {
+        (self.requested_size() as f64 * self.reservation_factor).round() as u64
```

Comment:
It seems like `ceil()` might be safer than `round()`ing. I'm also not sure how to handle it, but checking for overflow in the conversion back to a `u64` seems helpful too. One way might be adding checks in the `Region::new()` constructor, say that the factor is >= 1.0 and that the reserved size won't overflow a u64.

Reply:
95696ef00636f16345ed388457407ffbc76e81bf removes the floating point math, and restricts the factor via an enum.

Reply:
I like it, much simpler.